src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -174,6 +179,74 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
         }
 
+        private void CheckArgumentNullability(MethodSymbol attributeConstructor, AnalyzedAttributeArguments arguments, DiagnosticBag diagnostics)
+        {
+            // Check constructor arguments
+            for (int i = 0; i < attributeConstructor.ParameterCount && i < arguments.ConstructorArguments.Arguments.Count; i++)
+            {
+                var parameter = attributeConstructor.Parameters[i];
+                var argument = arguments.ConstructorArguments.Argument(i);
```

Comment:
>Argument(i) [](start = 62, length = 11)

Are the arguments guaranteed to be in parameter order?

Reply:
I doubt it, named arguments can be given out of order and should be evaluated in the order written, not parameter order.

---
In reply to: [262145720](https://github.com/dotnet/roslyn/pull/33816#discussion_r262145720) [](ancestors = 262145720)